Source/RunActivity/Viewer3D/WebServices/SwitchPanel/SwitchOnPanelDefinition.cs

```diff
@@ -0,0 +1,336 @@
+                    break;
+
+                case ORTS.Common.Input.UserCommand.ControlForwards:
+                    init2(ORTS.Common.Input.UserCommand.ControlForwards, ORTS.Common.Input.UserCommand.ControlBackwards);
+                    break;
+
+                case ORTS.Common.Input.UserCommand.ControlHandbrakeFull:
+                    Train train = Viewer.PlayerLocomotive.Train;
+
+                    int handBrakeCount = 0;
+
+                    for (int i = 0; i < train.Cars.Count; i++)
+                    {
+                        if ((train.Cars[i] as MSTSWagon).HandBrakePresent)
+                            handBrakeCount++;
+                    }
+
+                    if (handBrakeCount > 0)
+                        init2(ORTS.Common.Input.UserCommand.ControlHandbrakeFull, ORTS.Common.Input.UserCommand.ControlHandbrakeNone);
+                    else
+                        init0(ORTS.Common.Input.UserCommand.ControlHandbrakeFull);
+                    break;
+
+                case ORTS.Common.Input.UserCommand.ControlBrakeHoseConnect:
+                    init2(ORTS.Common.Input.UserCommand.ControlBrakeHoseConnect, ORTS.Common.Input.UserCommand.ControlBrakeHoseDisconnect);
+                    break;
+
+                case ORTS.Common.Input.UserCommand.ControlRetainersOn:
+                    init2(ORTS.Common.Input.UserCommand.ControlRetainersOn, ORTS.Common.Input.UserCommand.ControlRetainersOff);
+                    break;
+
+                default:
+                    init1(userCommand);
+                    break;
+            }
+        }
+
+        // empty button
+        public void initEmpty()
+        {
+            NoOffButtons = 0;
+            UserCommand = new UserCommand[] { ORTS.Common.Input.UserCommand.GamePauseMenu };
+            Description = "";
+        }
+
+        // almost empty non functioning button
+        private void init0(UserCommand userCommand)
+        {
+            NoOffButtons = 0;
+            UserCommand = new UserCommand[] { userCommand };
+            Description = determineDescription(userCommand);
+        }
+
+        // 1 button
+        private void init1(UserCommand userCommand)
+        {
+            NoOffButtons = 1;
+            UserCommand = new UserCommand[] { userCommand };
+            Description = determineDescription(userCommand);
+        }
+
+        // 2 buttons
+        private void init2(UserCommand userCommandTop, UserCommand userCommandBottom)
+        {
+            NoOffButtons = 2;
+            UserCommand = new UserCommand[] { userCommandTop, userCommandBottom };
+            Description = determineDescription(userCommandTop);
+        }
+
+        private string determineDescription(UserCommand userCommand)
+        {
+            string description = "";
+
+            /*
+             * this code could be simplified, however the GetParticularString is necessar to make translations possible
+             */
```

Comment:
Yes, it might be part of the enum UserCommand, however then it ends up into another assembly. On Elvis Tower I tried to start a discussion for this problem: http://www.elvastower.com/forums/index.php?/topic/37156-translations-gnugettext/page__p__297487__hl__translations__fromsearch__1#entry297487. But got no answers. But I think the current implementation is ok.

Reply:
Okay, I'm no expert either, so let's keep it as is